backend/__tests__/api.test.js

```diff
@@ -9,85 +9,242 @@ let userId;
+        isStarted: false,
+      });
+    taskObjectId = taskRes.body._id;
+  });
+
+  it("Getting All Tasks (Valid token)", async () => {
+    const res = await request(app)
+      .get("/api/tasks")
+      .set("Authorization", `Bearer ${jwt}`);
+    expect(res.statusCode).toEqual(200);
+    expect(res.body).toEqual(
+      expect.arrayContaining([expect.objectContaining({ title: "Test Task" })])
+    );
+  });
+
+  it("Getting All Tasks (Invalid token)", async () => {
+    const res = await request(app)
+      .get("/api/tasks")
+      .set("Authorization", `Bearer `);
+    expect(res.statusCode).toEqual(401);
+    expect(res.body.message).toEqual("Not authorized, token failed");
+  });
+
+  it("Getting Tasks by Date Range (Valid Token)", async () => {
+    const res = await request(app)
+      .get("/api/tasks?start=20220401&end=20270407")
+      .set("Authorization", `Bearer ${jwt}`);
+    expect(res.statusCode).toEqual(200);
+    expect(res.body).toEqual(
+      expect.arrayContaining([expect.objectContaining({ title: "Test Task" })])
+    );
+  });
+
+  it("Getting Tasks By Date Range (Invalid token)", async () => {
+    const res = await request(app)
+      .get("/api/tasks?start=20220401&end=20270407")
+      .set("Authorization", `Bearer `);
+    expect(res.statusCode).toEqual(401);
+    expect(res.body.message).toEqual("Not authorized, token failed");
+  });
+
+  it("Getting Tasks by id (Valid Token)", async () => {
+    const res = await request(app)
+      .get(`/api/tasks/task/${taskObjectId}`)
+      .set("Authorization", `Bearer ${jwt}`);
+
+    expect(res.statusCode).toEqual(200);
+    expect(res.body).toEqual(
+      expect.arrayContaining([expect.objectContaining({ title: "Test Task" })])
+    );
+  });
+
+  it("Getting Tasks by id (Invalid Token)", async () => {
+    const res = await request(app)
+      .get(`/api/tasks/task/${taskObjectId}`)
+      .set("Authorization", `Bearer `);
+
+    expect(res.statusCode).toEqual(401);
+    expect(res.body.message).toEqual("Not authorized, token failed");
+  });
+});
+
+describe("Create Task Suite", () => {
+  it("Create Task (Valid Token)", async () => {
+    const res = await request(app)
+      .post("/api/tasks")
+      .set("Authorization", `Bearer ${jwt}`)
+      .send({
+        title: "Test Task",
+        user_id: userId,
+        description: "Test Task",
+        endDate: new Date().toISOString(),
+        isStarted: false,
+      });
+    expect(res.body.title).toEqual("Test Task");
+    expect(res.body.user_id).toEqual(userId);
+    expect(res.body.description).toEqual("Test Task");
+    expect(res.body.isStarted).toEqual(false);
+  });
+
+  it("Create Task (Invalid token)", async () => {
+    const res = await request(app)
+      .post("/api/tasks")
+      .set("Authorization", `Bearer `);
+    expect(res.statusCode).toEqual(401);
+    expect(res.body.message).toEqual("Not authorized, token failed");
+  });
+});
+
+describe("Create Feedback Suite", () => {
```

Comment:
I would also create an additional feedback test for when a user inputs incorrect information such as a number for `isStarted`